addons/stock_landed_costs/models/product.py

```diff
@@ -12,10 +12,17 @@ class ProductTemplate(models.Model):
     landed_cost_ok = fields.Boolean('Is a Landed Cost', help='Indicates whether the product is a landed cost.')
 
     def write(self, vals):
+        products_to_change = []
         for product in self:
-            if (('type' in vals and vals['type'] != 'service') or ('landed_cost_ok' in vals and vals['landed_cost_ok'] == False)) and product.type == 'service' and product.landed_cost_ok:
+            if (('type' in vals and vals['type'] != 'service') or ('landed_cost_ok' in vals and not vals['landed_cost_ok'])) and product.type == 'service' and product.landed_cost_ok:
                 if self.env['account.move.line'].search_count([('product_id', 'in', product.product_variant_ids.ids), ('is_landed_costs_line', '=', True)]):
                     raise UserError(_("You cannot change the product type or disable landed cost option because the product is used in an account move line."))
-                else:
-                    vals['landed_cost_ok'] = False
-            super(ProductTemplate, product).write(vals)
+                products_to_change.append(product.id)
```

Comment:
I think you can just add vals['landed_cost_ok'] = False here and return super().write(vals) at the end

Reply:
same `vals` for every one so I can't do this

Reply:
Why not since they mandatory have it to false otherwise and you are writing it's not a service nor a landed cost

Reply:
In the worst case, you just rewrite the same value IMO